lib/logger/lib/logger/config.ex

```diff
@@ -21,50 +19,74 @@ defmodule Logger.Config do
     :gen_event.call(Logger, @name, {:remove_translator, translator})
   end
 
-  def __data__() do
-    try do
-      :ets.lookup_element(@table, :data, 2)
-    rescue
-      ArgumentError ->
-        raise "cannot use Logger, the :logger application is not running"
-    else
-      nil ->
-        raise "cannot use Logger, the :logger application is not running"
+  def level do
+    %{level: level} = read_data!(:translation_data)
+    level
+  end
 
-      data ->
-        data
+  def compare_levels(level, level), do: :eq
+
+  def compare_levels(left, right) do
+    if level_to_number(left) > level_to_number(right), do: :gt, else: :lt
+  end
+
+  defp level_to_number(:debug), do: 0
+  defp level_to_number(:info), do: 1
+  defp level_to_number(:warn), do: 2
+  defp level_to_number(:error), do: 3
+
+  def translation_data do
+    read_data!(:translation_data)
+  end
+
+  def log_data(level) do
+    %{level: min_level} = config = read_data!(:log_data)
+
+    if compare_levels(level, min_level) != :lt do
+      %{thresholds: {counter, sync, discard}} = config
+      value = bump_counter(counter)
+
+      cond do
+        value >= discard -> {:discard, config}
+        value >= sync -> {:sync, config}
+        true -> {:async, config}
+      end
+    else
+      {:discard, config}
     end
   end
 
-  def deleted_handlers() do
+  def deleted_handlers do
     try do
       :ets.lookup_element(@table, :deleted_handlers, 2)
     rescue
-      ArgumentError ->
-        []
+      ArgumentError -> []
     end
   end
 
   def deleted_handlers(handlers) do
     :gen_event.call(Logger, @name, {:deleted_handlers, handlers})
   end
 
-  def new() do
-    tab = :ets.new(@table, [:named_table, :public, {:read_concurrency, true}])
-    true = :ets.insert_new(@table, [{:data, nil}, {:deleted_handlers, []}])
-    tab
+  def new do
+    {new_data(), new_counter()}
```

Comment:
Why are we passing the "data" here to always pattern match against `@table` later? (taking into account this is a private module).

Reply:
Yeah, we are just being assertive the correct data is being passed around. We could also just use it without checking. Either way is fine.